pkg/ipamd/datastore/data_store.go

```diff
@@ -1171,3 +1170,15 @@ func GetPrefixDelegationDefaults() (int, int, int) {
 
 	return numPrefixesPerENI, numIPsPerPrefix, supportedPrefixLen
 }
+
+// FindFreeableCidrs finds and returns IPs that are not assigned to Pods but are attached
+// to ENIs on the node.
+func (ds *DataStore) FindFreeableCidrs(eni string) ([]net.IPNet, error) {
+	var freeableCidrs []net.IPNet
+	if !ds.isPDEnabled {
```

Comment:
Yeah mainly to avoid the copying I just returned the IPs or prefixes based on the mode. Also for instance if PD is enabled and a pod belonging to SIP is deleted, I cleanup the IP from datastore and free from EC2 since we would need that space for a /28 when more pods are deployed. But I can take up option 1 since that would replace both the functions and take up Option 2 as a fast follow up.